log4j-docker/src/main/java/org/apache/logging/log4j/docker/DockerLookup.java

```diff
@@ -46,6 +46,15 @@
     private final Container container;
 
     public DockerLookup() {
+
+        try {
+            // docker check.
+            Loader.newCheckedInstanceOf("org.apache.logging.log4j.docker.DockerLookup", StrLookup.class);
```

Comment:
These checks are attempting to load the class they're in. They won't load properly when `SystemLoader.load()` tries to load the class itself due to the static references to optional classpath code. This check you've added is only useful for code outside this exact file. The same goes for the other classes you've added these to.